multibase-conv/main.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	multibase "github.com/multiformats/go-multibase"
+)
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Printf("usage: %s NEW-BASE CID...\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	newBase := os.Args[1]
+	cids := os.Args[2:]
+
+	for _, cid := range cids {
+		_, data, err := multibase.Decode(cid)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			os.Exit(1)
+		}
+
+		newCid, err := multibase.Encode(multibase.Encoding(newBase[0]), data)
```

Comment:
`multibase.Encoding(newBase[0])` can happen before the loop and stored in a variable.